examples/static_set/device_subsets_example.cu

```diff
@@ -0,0 +1,154 @@
+                                                    cuda::thread_scope_device,
+                                                    thrust::equal_to<key_type>,
+                                                    probing_scheme_type,
+                                                    storage_ref_type,
+                                                    Operator>;
+
+/// data to insert/query
+__device__ constexpr std::array<key_type, N> data = {1, 3, 5, 7, 9, 11, 13, 15, 17, 19};
+/// Empty slots are represented by reserved "sentinel" values. These values should be selected such
+/// that they never occur in your input data.
+key_type constexpr empty_key_sentinel = -1;
+
+template <typename WindowT>
+__global__ void initialize(WindowT* windows, std::size_t n, typename WindowT::value_type value)
+{
+  using T = typename WindowT::value_type;
+
+  auto const loop_stride = gridDim.x * blockDim.x;
+  auto idx               = blockDim.x * blockIdx.x + threadIdx.x;
+
+  while (idx < n) {
+    auto& window_slots = *(windows + idx);
+#pragma unroll
+    for (auto& slot : window_slots) {
+      new (&slot) T{value};
+    }
+    idx += loop_stride;
+  }
+}
+
+// insert a set of keys into a hash set using one cooperative group for each task
+template <typename Window, typename Size, typename Offset>
+__global__ void insert(Window* windows, Size* sizes, Offset* offsets)
+{
+  namespace cg = cooperative_groups;
+
+  auto const tile = cg::tiled_partition<cg_size>(cg::this_thread_block());
+  auto const idx  = (blockDim.x * blockIdx.x + threadIdx.x) / cg_size;
+
+  auto set_ref = ref_type<cuco::experimental::insert_tag>{
+    cuco::empty_key<key_type>{-1}, {}, {}, storage_ref_type{sizes[idx], windows + offsets[idx]}};
+
+  // Each cooperative_groups inserts all elements in `data` into its own subset
+  for (int i = 0; i < N; i++) {
+    set_ref.insert(tile, data[i]);
+  }
+}
+
+// insert a set of keys into a hash set using one cooperative group for each task
+template <typename Window, typename Size, typename Offset>
+__global__ void find(Window* windows, Size* sizes, Offset* offsets)
+{
+  namespace cg = cooperative_groups;
+
+  auto const tile = cg::tiled_partition<cg_size>(cg::this_thread_block());
+  auto const idx  = (blockDim.x * blockIdx.x + threadIdx.x) / cg_size;
+
+  auto set_ref = ref_type<cuco::experimental::find_tag>{
+    cuco::empty_key<key_type>{-1}, {}, {}, storage_ref_type{sizes[idx], windows + offsets[idx]}};
+
+  __shared__ int result;
+  if (threadIdx.x == 0) { result = 0; }
+  __syncthreads();
+
+  for (int i = 0; i < N; i++) {
+    auto const found = set_ref.find(tile, data[i]);
+    // Record if the inserted data has been found
+    atomicOr(&result, *found != data[i]);
+  }
+  __syncthreads();
+
+  if (threadIdx.x == 0) {
+    if (result == 0) { printf("Success! Found all inserted elements.\n"); }
+  }
+}
+
+/**
+ * @file device_subsets_example.cu
+ * @brief Demonstrates usage of the static_set device-side APIs.
```

Comment:
Can you provide an explanation about the usecase?

Reply:
Yeah, let's strive to have in the top of each example file a high-level description of what the example does and what it is demonstrating.

Reply:
Sure, will do. I make this PR ready for review mainly to collect comments on how to handle `window_extent` asserts: https://github.com/NVIDIA/cuCollections/pull/346#discussion_r1295266470

Reply:
Docs and comments added.